hs_collection_resource/templates/pages/collectionresource.html

```diff
@@ -427,47 +447,139 @@ <h4 class="modal-title" id="save-collection-warning-title">Deleted Resources</h4
             }
         }
 
-
-        function remove_collection_item(res_id) {
+        function remove_collection_item_popup(res_id) {
             // get row obj is being removed from collection
             var removed_row_from_colletion = $('#collection-table').DataTable().row('#' + res_id);
 
             if(removed_row_from_colletion.data()[5].toLowerCase().indexOf("none") >= 0) {
-                $('#save-collection-btn-cancel').off("click");
-                $('#save-collection-btn-ok').off("click").on("click", function(){
-                    removed_row_from_colletion.remove().draw(false);
+                $('#remove-collection-btn-cancel').off("click");
+                $('#remove-collection-btn-ok').off("click").on("click", function(){
+                    remove_collection_item_ajax(res_id, false);
+                });
+
+                waning_message = "You have NO PERMISSION over the resource you are about to remove, which means you CAN NOT add it back later.";
+
+                $('#remove-collection-warning-body').text(waning_message);
+                $('#remove-collection-warning').modal('show');
+            }
+            else if(removed_row_from_colletion.data()[4].toLowerCase().indexOf("shareable") < 0 &&
+            removed_row_from_colletion.data()[5].toLowerCase().indexOf("owner") < 0) {
```

Comment:
Add a comment describing to what each index corresponds to in the table, or use variables that can identify them